README
NAME

    Dist::Zilla::Plugin::ContributorsFromGit - Populate your 'CONTRIBUTORS'
    POD from the list of git authors

VERSION

    This document describes version 0.016 of
    Dist::Zilla::Plugin::ContributorsFromGit - released January 18, 2015 as
    part of Dist-Zilla-Plugin-ContributorsFromGit.

SYNOPSIS

        ; in your dist.ini
        [ContributorsFromGit]
    
        ; in your weaver.ini
        [Contributors]

DESCRIPTION

    This plugin makes it easy to acknowledge the contributions of others by
    populating a %PodWeaver stash with the unique list of all git commit
    authors reachable from the current HEAD.

OVERVIEW

    On collecting the unique list of reachable commit authors from git, we
    search and remove any git authors from the list of authors Dist::Zilla
    knows about. We then look for a stash named %PodWeaver; if we don't
    find one then we create an instance of Dist::Zilla::Stash::PodWeaver
    and register it with our zilla instance. Then we add the list of
    contributors (the filtered git authors list) to the stash in such a way
    that Pod::Weaver::Section::Contributors can find them.

    Note that you do not need to have the %PodWeaver stash created; it will
    be added if it is not found. However, your Pod::Weaver config (aka
    c<weaver.ini>) must include the Contributors section plugin.

 Dist::Zilla Phase

    This plugin runs during the BeforeBuild phase.

 Metadata Keys

    The list of contributors is also added to distribution metadata under
    the custom x_contributors key. (e.g. in META.yml, META.json, etc)

    If you have duplicate contributors because of differences in committer
    name or email you can use a .mailmap file to canonicalize contributor
    names and emails. See git help shortlog for details.

 Pod::Weaver::Section::Contributors is OPTIONAL

    Note that using the Pod::Weaver::Section::Contributors section is in no
    way mandated or necessitated by this package; if you wish to use it you
    must include the Contributors section in your Pod::Weaver configuration
    in the traditional fashion.

METACPAN CONTRIBUTOR MATCHING

    MetaCPAN <http://metacpan.org> will attempt to match a contributor
    address back to a PAUSE account. However, it (currently) can only do
    that if the contributor's email address is their PAUSEID@cpan.org
    address. There are two mechanisms for helping to resolve this, if your
    commits are not using this address.

    Both of these approaches have pros and cons that have been discussed at
    levels nearing the heat brought to any discussion of religion,
    homosexuality, or Chief O'Brien's actual rank at any ST:TNG convention.
    However, they both have the advantage of *working*, and through
    different modes of action. You are free to use one, both or neither.
    These are only important if you're not committing with your @cpan.org
    email address and want the MetaCPAN to link to your author page from
    the page of the package you contributed to.

 Using a .mailmap file

    See git help shortlog for help on how to use this. A .mailmap file must
    be maintained in each repository using it.

 Globally, via the authors mapping

    This package contains a YAML file containing a mapping of @cpan.org
    author addresses; this list is consulted while building the
    contributors list, and can be used to replace a non-cpan.org address
    with one.

    To add to or modify this mapping, please feel free to fork, add your
    alternate email addresses to share/author-emails.yaml, and submit a
    pull-request for inclusion. It'll be merged and released; as various
    authors update their set of installed modules and cut new releases, the
    mapping will appear.

SEE ALSO

    Please see those modules/websites for more information related to this
    module.

      * Pod::Weaver::Section::Contributors

      * Dist::Zilla::Stash::PodWeaver

      *
      http://www.dagolden.com/index.php/1921/how-im-using-distzilla-to-give
      -credit-to-contributors/

SOURCE

    The development version is on github at
    http://https://github.com/RsrchBoy/Dist-Zilla-Plugin-ContributorsFromGi
    t and may be cloned from
    git://https://github.com/RsrchBoy/Dist-Zilla-Plugin-ContributorsFromGit
    .git

BUGS

    Please report any bugs or feature requests on the bugtracker website
    https://github.com/RsrchBoy/Dist-Zilla-Plugin-ContributorsFromGit/issue
    s

    When submitting a bug or request, please include a test-file or a patch
    to an existing test-file that illustrates the bug or desired feature.

AUTHOR

    Chris Weyl <cweyl@alumni.drew.edu>

 I'm a material boy in a material world

    Please note I do not expect to be gittip'ed or flattr'ed for this work,
    rather it is simply a very pleasant surprise. I largely create and
    release works like this because I need them or I find it enjoyable;
    however, don't let that stop you if you feel like it ;)

    Flattr this
    <https://flattr.com/submit/auto?user_id=RsrchBoy&url=https%3A%2F%2Fgith
    ub.com%2FRsrchBoy%2FDist-Zilla-Plugin-ContributorsFromGit&title=RsrchBo
    y's%20CPAN%20Dist-Zilla-Plugin-ContributorsFromGit&tags=%22RsrchBoy's%2
    0Dist-Zilla-Plugin-ContributorsFromGit%20in%20the%20CPAN%22>, gittip me
    <https://www.gittip.com/RsrchBoy/>, or indulge my Amazon Wishlist
    <http://bit.ly/rsrchboys-wishlist>... If you so desire.

CONTRIBUTORS

      * Brendan Byrd <Perl@ResonatorSoft.org>

      * David Golden <dagolden@cpan.org>

      * Graham Knop <haarg@haarg.org>

      * Mike Greb <mikegrb@cpan.org>

      * Randy Stauner <randy@magnificent-tears.com>

      * Tatsuhiko Miyagawa <miyagawa@bulknews.net>

      * Yanick Champoux <yanick@babyl.dyndns.org>

COPYRIGHT AND LICENSE

    This software is Copyright (c) 2012 by Chris Weyl.

    This is free software, licensed under:

      The GNU Lesser General Public License, Version 2.1, February 1999



Changes
Revision history for Dist-Zilla-Plugin-ContributorsFromGit

{{$NEXT}}

0.019     2017-08-14 13:44:31 CDT-0500
	* Correct mis-attribution of included foreign code

0.018     2017-08-13 21:07:22 CDT-0500
	* Fix tests for non-pty STDIN

0.017     2015-04-09 22:14:03 PDT-0700
	* Stop using Test::TempDir/Directory::Scratch; switch to
	  Test::TempDir::Tiny

0.016     2015-01-18 14:51:27 PST-0800
	* use Moose vs Reindeer; no need for *all* the antlers right now.
	  (issue #21)

0.015     2015-01-18 14:37:50 PST-0800
	* Use Test::File::SD instead of F::SD::ProjectDistDir (thanks yanick!)

0.014     2014-07-05 15:28:48 PDT-0700
	* merge pull request #20 (additional alternate author information)

0.013     2014-04-17 19:21:09 PDT-0700
	* properly *pass* our stopwords list to the PW StopWords **plugin**
	* document with more vigor that the PW Contributors section is optional,
	  and not required in any way by this plugin (issue #12)
	* make Dist::Zilla 5.013 the minimum required, to help resolve
	  META.{yml,json} problems (issue #9)
	* make author matching case-insensitive (closes #6, closes #11)

0.012     2014-04-17 14:45:19 PDT-0700
	* properly generate our stopwords list

0.011     2014-04-15 22:25:59 PDT-0700
	* bump Reindeer prereq version to 0.017, to avoid Try::Tiny pitfalls

0.010     2014-04-08 17:06:02 PDT-0700
	* add public interface validation tests
	* make all our attributes "private"
	* refactor documentation around how MetaCPAN handles contributor => PAUSE
	  ID matching
	* prep to break out our config email override into its own package

0.009     2014-04-08 00:45:48 PDT-0700
	* Put the authors mapping into a ShareDir yaml file

0.008     2014-04-07 14:15:05 PDT-0700
	* NO CODE CHANGES
	* Handle git identity crisis resulting in test failures a la
	  http://www.cpantesters.org/cpan/report/b6c182c2-be62-11e3-add5-ed1d4a243164

0.007     2014-04-03 22:00:00 PDT-0700
	* add support for author email mapping
	* Minimum supported Perl is now v5.10.1

0.006     2013-04-03 12:46:33 PDT-0700
	* add utf8 support (gh#4, thanks miyagawa)
	* ...and some (hopefully accurate) tests for said utf8 support

0.005     2013-02-28 06:00:00 PST-0800
	* ...and don't add an empty list of contributors (gh#2, thanks dagolden!)

0.004     2013-02-26 20:23:40 PST-0800
	* Add the list of contributors to META.* (gh#1, thanks dagolden!)

0.003     2012-12-27 21:42:43 CST6CDT
	* don't ABEND on no git, just skip tests and return early from the plugin

0.002     2012-12-02 13:52:34 PST8PDT
	* pod update -- note we run in the BeforeBuild phase
	* filter 'Your Name <you@example.com>' from the built list.  Really, we
	  need functionality to filter / consolidate, but this will work for now.

0.001     2012-11-25 01:41:41 PST8PDT
	* initial release


Makefile.PL
#
# This file is part of Dist-Zilla-Plugin-ContributorsFromGit
#
# This software is Copyright (c) 2012 by Chris Weyl.
#
# This is free software, licensed under:
#
#   The GNU Lesser General Public License, Version 2.1, February 1999
#

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.031.
use strict;
use warnings;

use 5.010000;

use ExtUtils::MakeMaker;

use File::ShareDir::Install;
$File::ShareDir::Install::INCLUDE_DOTFILES = 1;
$File::ShareDir::Install::INCLUDE_DOTDIRS = 1;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "Populate your 'CONTRIBUTORS' POD from the list of git authors",
  "AUTHOR" => "Chris Weyl <cweyl\@alumni.drew.edu>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::ShareDir::Install" => "0.06"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-ContributorsFromGit",
  "EXE_FILES" => [],
  "LICENSE" => "lgpl",
  "MIN_PERL_VERSION" => "5.010000",
  "NAME" => "Dist::Zilla::Plugin::ContributorsFromGit",
  "PREREQ_PM" => {
    "Dist::Zilla" => "5.013",
    "Dist::Zilla::Role::BeforeBuild" => 0,
    "Dist::Zilla::Role::MetaProvider" => 0,
    "Dist::Zilla::Role::RegisterStash" => 0,
    "Dist::Zilla::Stash::PodWeaver" => 0,
    "Encode" => 0,
    "File::ShareDir" => 0,
    "File::Which" => 0,
    "IPC::System::Simple" => 0,
    "List::AllUtils" => 0,
    "Moose" => 0,
    "MooseX::AttributeShortcuts" => 0,
    "MooseX::Types::Moose" => 0,
    "Path::Class" => 0,
    "YAML::Tiny" => 0,
    "aliased" => 0,
    "autobox::Core" => 0,
    "autobox::Junctions" => 0,
    "autodie" => 0,
    "utf8" => 0
  },
  "TEST_REQUIRES" => {
    "Directory::Scratch" => 0,
    "File::Spec" => 0,
    "File::chdir" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::CheckDeps" => "0.010",
    "Test::DZil" => 0,
    "Test::File::ShareDir" => 0,
    "Test::Moose::More" => 0,
    "Test::More" => "0.94",
    "Test::TempDir" => 0,
    "lib" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "VERSION" => "0.016",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Directory::Scratch" => 0,
  "Dist::Zilla" => "5.013",
  "Dist::Zilla::Role::BeforeBuild" => 0,
  "Dist::Zilla::Role::MetaProvider" => 0,
  "Dist::Zilla::Role::RegisterStash" => 0,
  "Dist::Zilla::Stash::PodWeaver" => 0,
  "Encode" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::ShareDir" => 0,
  "File::ShareDir::Install" => "0.06",
  "File::Spec" => 0,
  "File::Which" => 0,
  "File::chdir" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "IPC::System::Simple" => 0,
  "List::AllUtils" => 0,
  "Moose" => 0,
  "MooseX::AttributeShortcuts" => 0,
  "MooseX::Types::Moose" => 0,
  "Path::Class" => 0,
  "Test::CheckDeps" => "0.010",
  "Test::DZil" => 0,
  "Test::File::ShareDir" => 0,
  "Test::Moose::More" => 0,
  "Test::More" => "0.94",
  "Test::TempDir" => 0,
  "YAML::Tiny" => 0,
  "aliased" => 0,
  "autobox::Core" => 0,
  "autobox::Junctions" => 0,
  "autodie" => 0,
  "lib" => 0,
  "strict" => 0,
  "utf8" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}


